kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,185 @@
+		thenCompareGetPropertiesToPassedProperties();
+	}
+
+	@Test
+	public void shouldReturnStringWithGet() {
+		givenTheMapWith("testKey1", "testString1");
+		whenNetworkPropsIsCreated();
+		thenGetReturnsExpectedObjectWithKey("testKey1");
+	}
+
+	@Test
+	public void shouldReturnStringWithGetOpt() {
+		givenTheMapWith("testKey1", "testString1");
+		whenNetworkPropsIsCreated();
+		thenGetOptReturnsExpectedObjectWithKey("testKey1");
+	}
+
+	@Test
+	public void shouldReturnEmptyGetOptEmpty() {
+		givenTheMapWith("testKeyEmpty", "");
+		whenNetworkPropsIsCreated();
+		thenGetOptReturnsExpectedEmptyWithKey("testKeyEmpty");
+	}
+
+	@Test
+	public void shouldReturnEmptyGetOptNull() {
+		givenTheMapWith("testKeyNull", null);
+		whenNetworkPropsIsCreated();
+		thenGetOptReturnsExpectedEmptyWithKey("testKeyNull");
+	}
+
+	@Test
+	public void shouldReturnEmptyGetOptEmptyKey() {
+		givenTheMapWith("testKeyNull", null);
+		whenNetworkPropsIsCreated();
+		thenGetOptReturnsExpectedEmptyWithKey("");
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithCommaSeperatedString() {
+		givenTheMapWith("testKey-comma-seperated", "commaSeperated1,commaSeperated2,commaSeperated3");
+		whenNetworkPropsIsCreated();
+		thenReturnStringListFromGetStringList("testKey-comma-seperated");
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithRegularString() {
+		givenTheMapWith("testKey1", "testString1");
+		whenNetworkPropsIsCreated();
+		thenReturnStringFromGetStringList("testKey1");
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithNull() {
+		givenTheMapWith("testKeyNull", null);
+		whenNetworkPropsIsCreated();
+		thenReturnEmptyListFromGetStringList("testKeyNull");
+	}
+
+	@Test
+	public void shouldReturnOptionalStringListGetStringListOptWithCommaSeperatedString() {
+		givenTheMapWith("testKey-comma-seperated", "commaSeperated1,commaSeperated2,commaSeperated3");
+		whenNetworkPropsIsCreated();
+		thenReturnOptStringListOptFromGetStringList("testKey-comma-seperated");
+	}
+
+	@Test
+	public void shouldReturnOptionalStringListGetStringListOptWithString() {
+		givenTheMapWith("testKey1", "testString1");
+		whenNetworkPropsIsCreated();
+		thenReturnOptStringOptFromGetStringList("testKey1");
+	}
+
+	@Test
+	public void shouldReturnEmptyGetStringListOptWithNull() {
+		givenTheMapWith("testKeyNull", null);
+		whenNetworkPropsIsCreated();
+		thenReturnOptEmptyOptFromGetStringList("testKeyNull");
+	}
+
+	@Test
+	public void shouldReturnEmptyGetStringListOptWithNoKey() {
+		givenTheMapWith("testKeyNull", null);
+		whenNetworkPropsIsCreated();
+		thenReturnOptEmptyOptFromGetStringList("");
+	}
+
+	// Given //
+	public void givenTheFollowingProperties(Map<String, Object> properties) {
+		this.properties = properties;
+	}
+	
+	public void givenTheMapWith(String key, Object pair) {
+		properties = new HashMap<String, Object>();
+		properties.put(key, pair);
+	}
+	// End of Given //
+
+	// When //
+
+	public void whenNetworkPropsIsCreated() {
```

Comment:
IMO This is still a `given` i.e. is part of the test setup.

> In the GIVEN clause, you should express any kind of setup conditions that are necessary for this test to succeed. This encompasses hardware conditions as well as data, files, or records that must have a specific condition for the test to run.

So:

```suggestion
	public void giveNetworkPropsIsCreated() {
```

or

```suggestion
	public void giveNetworkPropsIsCreatedWith(Map<String, Object> props) {
```

Reply:
I think I disagree with this. In this step, we "configure the system" that we are going to test—by instantiating and applying the config?  ("by creating and configuring our object to test")

Thoughts?